File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/pigeon_implement/DeviceLogDelegateImpl.swift
//
//  DeviceLogDelegateImpl.swift
//  protocol_channel
//
//  Created by hedongyang on 2023/11/23.
//

import Foundation

class DeviceLogDelegateImpl: DeviceLogDelegate {
    
    static let shared = DeviceLogDelegateImpl()
    private init() {}
    
    private(set) var logStatus : Bool = false
    private(set) var logDirPath: String = ""
    var callbackProgress: BlockLogProgress?
    
    func listenLogStatus(status: Bool) throws {
        logStatus = status
    }
    
    func listenLogDirPath(dirPath: String) throws {
         logDirPath = dirPath
    }
    
    func callbackLogProgress(progress: Double) throws {
        callbackProgress?(progress)
    }
    
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/pigeon_implement/DeviceLogDelegateImpl.kt
package com.idosmart.pigeon_implement

import com.idosmart.pigeongen.api_device_log.DeviceLogDelegate
import com.idosmart.protocol_sdk.BlockLogProgress

internal class DeviceLogDelegateImpl: DeviceLogDelegate {

    companion object {
        @Volatile
        private var instance: DeviceLogDelegateImpl? = null
        fun instance(): DeviceLogDelegateImpl {
            return instance ?: synchronized(this) {
                instance ?: DeviceLogDelegateImpl().also { instance = it }
            }
        }
    }

    var logStatus: Boolean = false
    var logDirPath: String = ""
    var callbackProgress: BlockLogProgress? = null

    override fun listenLogStatus(status: Boolean) {
        logStatus = status
    }

    override fun listenLogDirPath(dirPath: String) {
        logDirPath = dirPath
    }

    override fun callbackLogProgress(progress: Double) {
        callbackProgress?.let {
            it(progress)
        }
    }

}